src/main/java/gregtech/api/metatileentity/multiblock/MultiblockControllerBase.java

```diff
@@ -349,13 +350,14 @@ public void checkStructurePattern() {
             this.setFlipped(context.neededFlip());
             parts.sort(Comparator.comparing(it -> multiblockPartSorter().apply(((MetaTileEntity) it).getPos())));
             Map<MultiblockAbility<Object>, List<Object>> abilities = new HashMap<>();
-            for (IMultiblockPart multiblockPart : parts) {
-                if (multiblockPart instanceof IMultiblockAbilityPart) {
-                    @SuppressWarnings("unchecked")
-                    IMultiblockAbilityPart<Object> abilityPart = (IMultiblockAbilityPart<Object>) multiblockPart;
-                    List<Object> abilityInstancesList = abilities.computeIfAbsent(abilityPart.getAbility(),
-                            k -> new ArrayList<>());
-                    abilityPart.registerAbilities(abilityInstancesList);
+            for (IMultiblockPart part : parts) {
+                if (part instanceof IMultiblockAbilityPart abilityPart) {
+                    List<MultiblockAbility> abilityList = abilityPart.getAbilities();
+                    for (MultiblockAbility ability : abilityList) {
+                        List abilityInstancesList = abilities.computeIfAbsent(ability,
+                                k -> new ArrayList<>());
+                        abilityInstancesList.addAll(abilityPart.registerAbilities(ability));
+                    }
```

Comment:
What happens here if a single multiblock part has `IMPORT_ITEMS` and `IMPORT_FLUIDS`, but the multiblock only allows one of these in the structure at that position. Should the part be allowed or rejected?